### PR TITLE
Week 4

### PR DESCRIPTION
**Pull request by Vlad4Per**

### CODE REVIEW

**Review by Kamil-Sabbagh — APPROVED** [review]
{{< hint danger >}}
**Feedback**

Overall, the report demonstrates substantial progress and integration of external feedback. However, several areas need improvement to meet the grading criteria fully.

**Project Focus & Approach (1/2)**
- The project's overall goal and relevance are clear.
- The specific focus of the week's work are lacking
- The importance of this week's work is justified.
- Methods and approaches used this week are described.
- The justification for choices made is somewhat vague.

**Results & Analysis (1/2)**
- Findings are presented clearly.
- Analysis of results is inadequate, with insufficient detail on the testing performed and metrics used.
- Implications for the overall project are discussed.

**Conclusions & Next Steps (0.5/1)**
- Key takeaways are summarized.
- Next steps are outlined but lack detail.

**Challenges & Solutions (0.5/1)**
- Challenges encountered are identified.
- Solutions to challenges are described.
- Reflection on lessons learned is minimal.

**Progress Review Session with TA (2/2)**

**Communication & Presentation (1.5/2)**
- The report is well-organized and visually appealing.
- Writing is clear and concise.
- Visuals support the text effectively.
- Proper citation and referencing are followed.

**Total: 6.5/10**

Suggestions:
- Provide more detailed analysis connecting results to objectives and specify testing methods and metrics.
- Include more specific next steps.
- Reflect more on the lessons learned from challenges.
- Ensure to mention and incorporate feedback from the TA progress review session.
{{< /hint >}}

**Comment by Vlad4Per** [comment]
Week 5 report added

File: content/docs/2024/QML_selection_service/week4.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
```
Comment (Kamil-Sabbagh) [inline content/docs/2024/QML_selection_service/week4.md:1]: {{< hint danger >}}
Feedback

Overall, the report demonstrates substantial progress and integration of external feedback. However, several areas need improvement to meet the grading criteria fully.

Project Focus & Approach (1/2)

The project's overall goal and relevance are clear.
The specific focus of the week's work are lacking
The importance of this week's work is justified.
Methods and approaches used this week are described.
The justification for choices made is somewhat vague.
Results & Analysis (1/2)

Findings are presented clearly.
Analysis of results is inadequate, with insufficient detail on the testing performed and metrics used.
Implications for the overall project are discussed.
Conclusions & Next Steps (0.5/1)

Key takeaways are summarized.
Next steps are outlined but lack detail.
Challenges & Solutions (0.5/1)

Challenges encountered are identified.
Solutions to challenges are described.
Reflection on lessons learned is minimal.
Progress Review Session with TA (2/2)

Communication & Presentation (1.5/2)

The report is well-organized and visually appealing.
Writing is clear and concise.
Visuals support the text effectively.
Proper citation and referencing are followed.
Total: 6.5/10

Suggestions:

Provide more detailed analysis connecting results to objectives and specify testing methods and metrics.
Include more specific next steps.
Reflect more on the lessons learned from challenges.
Ensure to mention and incorporate feedback from the TA progress review session.
{{< /hint >}}

File: content/docs/2024/QML_selection_service/week5.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+---
```
Comment (Kamil-Sabbagh) [inline content/docs/2024/QML_selection_service/week5.md:1]: {{< hint danger >}}
**Feedback**

The report provides a thorough update on the progress made in Week 5. It effectively covers feedback collection, analysis, and subsequent roadmap development. However, the sections on results analysis and connection to overall project goals lack sufficient depth. Also, the challenges and solutions section could be more detailed.

**Grade:**

- **Project Focus & Approach:** 1.75/2
  - Clearly articulates the project's overall goal and its relevance (0.5 points).
  - Defines the specific focus of the week's work (0.25 points).
  - Justifies the importance of this week's work (0.25 points).
  - Describes the methods, tools, or approaches used (0.5 points).
  - Lacks detailed justification of choices made (0.25 points).

- **Results & Analysis:** 1.25/2
  - Presents findings or outcomes clearly (0.5 points).
  - Analyzes results, drawing connections to objectives (0.5 points).
  - Does not adequately discuss implications for the overall project (0.25 points).

- **Conclusions & Next Steps:** 0.75/1
  - Summarizes key takeaways (0.5 points).
  - Outlines actionable next steps (0.25 points).

- **Challenges & Solutions:** 0.5/1
  - Identifies obstacles encountered (0.25 points).
  - Describes how challenges were addressed (0.25 points).
  - Missing reflection on lessons learned (0 points).

- **Progress Review Session with TA:** 1.5/2
  - Actively participates and demonstrates understanding of feedback (0.5 points).
  - Incorporates feedback into the report (1 point).

- **Communication & Presentation:** 1.5/2
  - The report is well-organized and easy to follow (0.5 points).
  - Writing is clear and appropriate (0.5 points).
  - Effective use of visuals (0.5 points).

**Total:** 7.25/10
{{< /hint >}}

File: content/docs/2024/QML_selection_service/week6.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (Kamil-Sabbagh) [inline content/docs/2024/QML_selection_service/week6.md:1]: {{< hint danger >}}
**Feedback**

The report provides a comprehensive overview of the progress made this week, including detailed descriptions of achievements, challenges, and next steps. However, the analysis of the results and their implications on the overall project is somewhat lacking. More detail on the lessons learned and reflections on the challenges would strengthen the report.

**Grade:**

- **Project Focus & Approach:** 1.5/2
  - Clearly articulates the project's overall goal and its relevance (0.5 points).
  - Defines the specific focus of the week's work (0.25 points).
  - Justifies the importance of this week's work (0.25 points).
  - Describes the methods, tools, or approaches used (0.5 points).
  - Lacks detailed justification of choices made (0 points).

- **Results & Analysis:** 1.25/2
  - Presents findings or outcomes clearly (0.5 points).
  - Analyzes results, drawing connections to objectives (0.5 points).
  - Does not adequately discuss implications for the overall project (0.25 points).

- **Conclusions & Next Steps:** 0.75/1
  - Summarizes key takeaways (0.5 points).
  - Outlines actionable next steps (0.25 points).

- **Challenges & Solutions:** 0.5/1
  - Identifies obstacles encountered (0.25 points).
  - Describes how challenges were addressed (0.25 points).
  - Missing reflection on lessons learned (0 points).

- **Progress Review Session with TA:** 1.5/2
  - Actively participates and demonstrates understanding of feedback (0.5 points).
  - Incorporates feedback into the report (1 point).

- **Communication & Presentation:** 1.5/2
  - The report is well-organized and easy to follow (0.5 points).
  - Writing is clear and appropriate (0.5 points).
  - Effective use of visuals (0.5 points).

**Total:** 7/10
{{< /hint >}}

**Review by Kamil-Sabbagh — APPROVED** [review]

**Pull request merged**